public/js/logic.js

```diff
@@ -0,0 +1,74 @@
+function fetchNode(url, data) {
```

Comment:
what is the point of this function? you can use fetch directly! you are not abstracting anything here